### ISSUE TITLE
64bit connection id may break slow query detail

### ISSUE THREAD

**Issue by breeswish** [pingcap-incubator/tidb-dashboard#791]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

**What did you do?**

TiDB generates 64bit connection id since https://github.com/pingcap/tidb/pull/17649. In some cases, the slow query detail page cannot correctly handle this case due to JavaScript max int limitation.

**What did you expect to see?**

Slow query detail work correctly.

**What did you see instead?**

Slow query detail may throw error.

**What version of TiDB Dashboard are you using (`./tidb-dashboard --version`)?**

**Comment by baurine** [comment]
Does it work by using string type in the response?

**Comment by unbyte** [comment]
https://caniuse.com/?search=BigInt

**Comment by kennytm** [comment]
TBH I'd prefer using [`json-bigint`](https://www.npmjs.com/package/json-bigint) + `bignumber.js` than changing the JSON type to string...

Or we could restrict the Global connection ID to use only 53 bits 🙃, reducing the range from 22(server ID) + 40(local conn) to 18(server ID) + 34(local conn).

**Comment by breeswish** [comment]
Prefer json-bigint +1

We'd better not changing the global connection ID bit structure since this is a fault of JavaScript instead of connection ID.

**Comment by breeswish** [comment]
Some discovery:

- For parse usage, occurs when receiving server response, this problem can be gracefully handled by customizing [`transformResponse`](https://github.com/axios/axios/blob/6642ca9aa1efae47b1a9d3ce3adc98416318661c/lib/defaults.js#L57) and introducing json-bigint.

- For stringify usage, occurs when generating server request, looks like there is no way to use bignumber or json-bigint (?). The request is generated by the swagger client. It accepts TypeScript `number` signatures, and then use [`url.format`](http://nodejs.cn/api/url/url_format_urlobject.html) to generate the actual URL. `url.format` internally use [`querystring.stringify`](https://nodejs.org/api/querystring.html#querystring_querystring_stringify_obj_sep_eq_options), which only accepts primitive types instead of a bignumber object. Any thoughts?

**Comment by kennytm** [comment]
can we map the BigInt toString before passing to swagger?

**Comment by breeswish** [comment]
> can we map the BigInt toString before passing to swagger?

@kennytm  Yes. However this loses its advantage. Public interfaces are no longer semantically correct, server side API and Swagger specs both accept `string` instead of `uint64 for` fields that should be `uint64`.

**Comment by kennytm** [comment]
(filed nodejs/node#36080 for bigint support in querystring)

(but why are we using nodejs?)

**Comment by breeswish** [comment]
> (filed [nodejs/node#36080](https://github.com/nodejs/node/issues/36080) for bigint support in querystring)
> 
> (but why are we using nodejs?)

We are using Webpack, so that in addition to NodeJs support, we also need to expect browser polyfills to work correctly.

According to https://medium.com/@sanchit3b/how-to-polyfill-node-core-modules-in-webpack-5-905c1f5504a0, url and querystring will be mapped to corresponding module in npm, e.g. https://www.npmjs.com/package/querystring

**Comment by kennytm** [comment]
ok. looks like even if we get nodejs/node#36080 fixed it will take a long time for us to support older nodejs versions in other dev environments 😅

**Comment by unbyte** [comment]
Since https://github.com/tc39/proposal-bigint is a very new feature, neither the surrounding ecology nor Polyfill is mature. Once bigint is used, it needs to introduce a lot of additional processing code to pollute the workflow and code. 
Therefore, I prefer to use string. Anyway, we did not change the nature of its numerical value. We only converted it into string for the display in frontend. This conversion is transparent to users.
